example/include/docca/issue_50.hpp

```diff
@@ -0,0 +1,578 @@
+//
+// Copyright (c) 2016-2019 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef EXAMPLE_ISSUE_50
+#define EXAMPLE_ISSUE_50
+
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <type_traits>
+
+namespace example {
+
+/** A dynamic buffer providing sequences of variable length.
+
+    A dynamic buffer encapsulates memory storage that may be
+    automatically resized as required, where the memory is
+    divided into two regions: readable bytes followed by
+    writable bytes. These memory regions are internal to
+    the dynamic buffer, but direct access to the elements
+    is provided to permit them to be efficiently used with
+    I/O operations.
+
+    The implementation uses a sequence of one or more byte
+    arrays of varying sizes to represent the readable and
+    writable bytes. Additional byte array objects are
+    appended to the sequence to accommodate changes in the
+    desired size. The behavior and implementation of this
+    container is most similar to `std::deque`.
+
+    Objects of this type meet the requirements of <em>DynamicBuffer</em>
+    and have the following additional properties:
+
+    @li A mutable buffer sequence representing the readable
+    bytes is returned by @ref data when `this` is non-const.
+
+    @li Buffer sequences representing the readable and writable
+    bytes, returned by @ref data and @ref prepare, may have
+    length greater than one.
+
+    @li A configurable maximum size may be set upon construction
+    and adjusted afterwards. Calls to @ref prepare that would
+    exceed this size will throw `std::length_error`.
+
+    @li Sequences previously obtained using @ref data remain
+    valid after calls to @ref prepare or @ref commit.
+
+    @tparam Allocator The allocator to use for managing memory.
+*/
+template<class Allocator>
+class basic_issue_50
+#if ! BOOST_BEAST_DOXYGEN
+: private boost::empty_value<Allocator>
```

Comment:
Do we really need all this? Please trim it all the way down to the smallest declaration which reproduces the defect. The less code and comments, the faster the doc toolchain runs.